parser.go

```diff
@@ -20,18 +20,21 @@ func ParseLink(line string) map[string]string {
 	m := make(map[string]string)
 
 	// Regex to extract title, link, and description
-	re := regexp.MustCompile(`(?m)(^- \[([^]]+)\]\(([^)]+)\) ?-? ?(.*)?)?`)
+	re := regexp.MustCompile(`(?m)(^- \[([^]]+)\]\(([^)]+)\) ?-? ?(.*)?)?(^!\[\]\(([^)]+)\))?`)
```

Comment:
It should be a separate function for this `ParseImageLink` for example.

Reason is that now if I say parse https://github.com/learn-anything/books this file and it will contain an image embed anywhere, the function would return me both the links and the images it found (I just want the links).

Reply:
Maybe we can achieve the same with this method too, not sure. But think in what contexts can this library be used.